### PR TITLE
apftool-rs: init at 0-unstable-2024-01-05

### PR DESCRIPTION
**Pull request by colemickens**
Package this tool: https://github.com/suyulin/apftool-rs

at an unstable revision.

It's the most reliable tool I've found for unpacking various rockchip images.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by KiaraGrouwstra** [comment]
Result of `nixpkgs-review pr 354398` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>apftool-rs</li>
  </ul>
</details>

**Comment by KiaraGrouwstra** [comment]
code looks good and the package works, but could you format it as per CI?

**Review by KiaraGrouwstra — APPROVED** [review]

File: pkgs/by-name/ap/apftool-rs/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "apftool-rs";
+  version = "unstable-2024-01-05";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ap/apftool-rs/package.nix:9]:
```suggestion
  version = "0-unstable-2024-01-05";
```
`version` must always start with a digit.

File: pkgs/by-name/ap/apftool-rs/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0-unstable-2024-01-05";
+
+  src = fetchFromGitHub {
+    owner = "suyulin";
+    repo = "apftool-rs";
+    rev = "92d8a1b88cb79a53f9e4a70fecee481710d3565b";
+    hash = "sha256-0+eKxaLKZBRLdydXxUbifFfFncAbthUn7AB8QieWaXM=";
+  };
+
+  cargoHash = "sha256-tijhuiuJhOirnRdDXRNafh3SxBmX65zPE5HJnHBUj30=";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ap/apftool-rs/package.nix:18]:
```suggestion
  cargoHash = "sha256-6lYokd0jwpBWCQ+AbN6ptZYXGcy41GHPbnTELUjPbyA=";
```
Is the value I get.

Comment (FliegendeWurst) [reply pkgs/by-name/ap/apftool-rs/package.nix:18]: It probably changed due to the `version` change.

**Pull request merged**